### PR TITLE
paramdigger: Add initial header guesser implementation

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
Related to zaproxy/zaproxy#7140
Signed-off-by: ArkaprabhaChakraborty <chakrabortyarkaprabha998@gmail.com>

### CODE REVIEW

**Comment by ArkaprabhaChakraborty** [comment]
Will be adding more things :)

**Comment by ArkaprabhaChakraborty** [comment]
This would have another sort of naming change I feel :). UrlUtils -> Utils :)

**Comment by ArkaprabhaChakraborty** [comment]
@thc202 @kingthorin This PR is ready. For this PR These are the intended changes that I wish to keep.

File: addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/Utils.java [hunk 1]
```diff
@@ -34,8 +34,16 @@
...
-public class UrlUtils {
-    private static Logger logger = LogManager.getLogger(UrlUtils.class);
+public class Utils {
+    /** The allowed Methods for the param digger add-on. */
+    public enum Method {
+        GET,
+        POST,
+        XML,
+        JSON,
+    }
```
Comment (thc202) [inline addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/Utils.java:44]: Move it to its own file, `Utils` does not seem the right place for it.

File: addOns/paramdigger/src/test/java/org/zaproxy/addon/paramdigger/UrlGuesserUnitTest.java [hunk 2]
```diff
@@ -85,8 +85,7 @@ private static String encodeUrl(String value) {
     @BeforeEach
     void init() throws Exception {
         setUpZap();
-        try (InputStream is =
-                UrlUtilsUnitTest.class.getResourceAsStream("/" + TEST_WORDLIST_FILE)) {
+        try (InputStream is = UtilsUnitTest.class.getResourceAsStream("/" + TEST_WORDLIST_FILE)) {
```
Comment (thc202) [inline addOns/paramdigger/src/test/java/org/zaproxy/addon/paramdigger/UrlGuesserUnitTest.java:88]: Use the own class.

**Comment by ArkaprabhaChakraborty** [comment]
Should be mergable now :)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**